client/consensus/grandpa/interfaces.go

```diff
@@ -26,13 +26,19 @@ type ForkTree[H comparable, N constraints.Unsigned] interface {
 	PendingChanges() []PendingChange[H, N]
 }
 
-// PublicKey interface
-type PublicKey interface {
-	Verify(msg, sig []byte) (bool, error)
-	Encode() []byte
-	Decode([]byte) error
-	Hex() string
-}
+type Telemetry interface{}
 
-type Telemetry interface {
+// AuxStore is part of the substrate backend.
+// Provides access to an auxiliary database.
+//
+// This is a simple global database not aware of forks. Can be used for storing auxiliary
+// information like total block weight/difficulty for fork resolution purposes as a common use
+// case.
+type AuxStore interface {
```

Comment:
I don't think this belongs in here.  It's located in `client/api` in substrate, so we can assume this will be used elsewhere.

Reply:
I remember Quentin doing that and a lot of people not liking it.  I think it makes sense to do it contextually, like say if we only need a portion of another package interface.